### DISCUSSION TITLE
Default Midi Note / Gate  Length or on the fly length adjustment

### DISCUSSION THREAD

**Discussion by sfjtdr** [SynthstromAudible/DelugeFirmware#1881] [Ideas]
Hi everyone. 
Many synth's and drum machines when fast successive steps are entered on the Deluge, only read as one long gate

For midi (or any sequencing really), It would be nice if there was a default setting for note / gate length. 
Or, a way to quickly adjust gate / note length, like how adjusting the velocity per pad already works, whereas every pad is the same value after after you've made a change.

I know that you can zoom in _(because the gate length changes dynamically with  zoom)_ and do cross screen as a work around, but that is tedious. Very tedious. Breaks the flow.

Pulsar-23 sequencing would benefit greatly from this. Staccato performances would be easier. Improvisation would be quicker and more dynamic. Almost every sequencer has a note length setting. 

this thread on the Deluge community page describes it well 
[https://forums.synthstrom.com/discussion/3204/change-default-note-length](https://forums.synthstrom.com/discussion/3204/change-default-note-length)

**Comment by duncanrmi** [comment]
I have a problem with the default note length when using the deluge to control (e.g.) a NDLR.
seems the default note length on the deluge at the default zoom level is four steps long a 1/64th note, which the NDLR hears as four separate note-ons of equal duration*, causing it to strum when I don't want it to.
in short, the default note length is determined by the zoom level.
not what all of us want.
editing these notes by zooming in & removing three of the pads is monumentally tedious. I'd like a global setting for the default note length.

(* this in itself is a bit mysterious, & I need to check with midi-monitor or something to see if there really are four note-ons)

**Reply by seangoodvibes** [reply]
It definitely should not be sending 4 note on's. Maybe the problem is on the receiving end?

**Reply by duncanrmi** [reply]
nope- see below.

**Comment by sfjtdr** [comment]
I think 4 ‘notes’ worth of gated length is what he meant. Not individual instances of a note. Maybe I’m wrong. Regardless, no default midi note length or on the fly adjustment is still a huge miss for me. I’ve resorted to another sequencer (torso t1) for certain instruments of mine. It’s one of the main knobs on that one. ‘Note length’. I’m actually shocked the community has gone this far with so many awesome and crazy features, and not done this. It seems so rudimentary. I wish I knew how to code. Sent from my iPhoneOn Oct 19, 2024, at 8:46 PM, Sean Ditny ***@***.***> wrote:﻿
It definitely should not be sending 4 note on's. Maybe the problem is on the receiving end?

**Reply by seangoodvibes** [reply]
Tbh I think this is easy to implement but harder to implement well.

You referenced the velocity adjustment as a good example.

I actually think the velocity adjustment is annoying most of the time, definitely for kits anyway.

Then first time you enter notes, sure, but the fact that every time you adjust a notes velocity afterwards it affects the input velocity of every other note afterwards is really frustrating.

It's unpredictable and easy to forget what velocity you're going to get for the next note you enter.

So while this is an easy request relatively speaking, we need a good solution and user experience if we added this.

Questions to consider:
- does it impact all notes in a synth?
- should it only impact all notes in a row?
- how can you reset it to normal so that new notes behave as per usual?
- how do you activate it so that notes adopt the length you want?

**Comment by duncanrmi** [comment]
definitely a number of note-on/note-off events per clip pad, the number depending on the resolution.
so at 128th notes, with a single pad illuminated, I see two note-ons followed by two note-offs. this is using the usb midi into a mac running midi-monitor.
the NDLR sees this as separate events, which is the problem, though at the 128th setting they seem to be too close together to register as separate events.
no, it should not be doing this. the gate time of a note event should be defined by the interval between a single note-on & the corresponding note-off, or else a second note-on of zero velocity. that's NOT what's happening, according to midi-monitor or the NDLR.

at default resolution on the deluge, I'm experiencing four note-ons far enough apart to hear as separate events if the target is short enough. what I do is zoom right in & turn the event off, back on again, so it's now a single pad. at 128th note resolution, this is short enough to get the NDLR to behave how I want, but it's still wrong, isn't it?

a different midi-monitor product on my work laptop, & a different deluge (to be fair...) gives me the expected results instead.
weird, eh?
![deluge op midiview](https://github.com/user-attachments/assets/1bd78f13-f6d5-4d46-909d-5a120aabf960)

**Reply by duncanrmi** [reply]
I'll post summat from the other deluge & the midi monitor on the mac later.

**Reply by duncanrmi** [reply]
uh... no need. so at 8th note resolution, by way of an experiment, I set ONE midi note on a pad, & extended its duration so that the next pad is half-lit. long note, right? no. two notes:
![deluge op midiview 2](https://github.com/user-attachments/assets/a927ed30-f969-4ae3-b652-0e92edb00d94)

& each additional pad of "I only wanted to increase the gate length" adds another note-on/note-off.

**Reply by duncanrmi** [reply]
this is at eighth-note resolution, one pad + three 'extend the note' pads, i.e. a single note with 1/2 note gate time:

![deluge op midiview 3](https://github.com/user-attachments/assets/b5fb8663-6579-4457-baf0-58145b351557)

**Reply by duncanrmi** [reply]
going to bug-report this.
